src/static/helpers/slasHelper.ts

```diff
@@ -260,7 +314,36 @@ export async function loginRegisteredUserB2C(
   }
 
   const authResponse = getCodeAndUsidFromUrl(redirectUrlString);
+  // using slas private client
+  if (credentials.clientSecret) {
+    /* istanbul ignore next */
+    if (isBrowser) {
+      // we intentionally have warning here
+      // eslint-disable-next-line
+      console.warn(warningSlasSecretMsg);
+    }
+
+    const authHeaderIdSecret = `Basic ${stringToBase64(
+      `${slasClient.clientConfig.parameters.clientId}:${credentials.clientSecret}`
+    )}`;
+
+    const optionsToken = {
+      headers: {
+        Authorization: authHeaderIdSecret,
+      },
+      body: {
+        grant_type: 'authorization_code_pkce',
+        code_verifier: codeVerifier,
+        code: authResponse.code,
+        client_id: slasClient.clientConfig.parameters.clientId,
+        redirect_uri: parameters.redirectURI,
+        usid: authResponse.usid,
+      },
+    };
+    return slasClient.getAccessToken(optionsToken);
+  }
```

Comment:
I think we can reuse the same optionsToken as the public client. When logging in a registered user, only difference in requirements between a public and private client is whether we need to include the auth header with the client Id and secret.

We still want the other parameters like siteId to be set in the body.

Reply:
Another way to think about it is that since the grant type here in both cases is authorization_code_pkce, the parameters in the body should be identical.